cpp/include/cudf/io/experimental/cudftable.hpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cudf/io/types.hpp>
+#include <cudf/packed_types.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/export.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <utility>
+
+namespace CUDF_EXPORT cudf {
+namespace io::experimental {
+
+/**
+ * @file
+ * @brief CUDFTable binary format APIs for serialization and deserialization
```

Comment:
For my education, are we inventing a new IO format?

Reply:
We are, unfortunately. The intent is for it to be as lean as possible on the compute side. Possible uses are caching and spilling.
This might get superseeded by Feather support, but that's a lot more effort to implement and this lets us experiment with a "raw data" format.